### ISSUE TITLE
Rule to disallow empty line before/after class?

### ISSUE THREAD

**Issue by brunolm** [palantir/tslint#3968]
I want to know if there is a lint rule for this

not allowed 
- empty line after class declaration line, before first element
- empty line before class end, after last element in class

wrong
```ts
class Foo {

  constructor() {
  }

}
```

correct
```ts
class Foo {
  constructor() {
  }
}
```

Is there a rule to do this? Or could I maybe suggest it?

**Comment by brunolm** [comment]
![image](https://user-images.githubusercontent.com/706078/41501339-f3523cae-7178-11e8-8257-1e5b2d7226c0.png)

```ts
// tslint:disable:max-classes-per-file

import * as Lint from 'tslint';
import * as ts from 'typescript';

export class Rule extends Lint.Rules.AbstractRule {
  public static failureMessage = 'padding space in classes not allowed';

  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
    return this.applyWithWalker(new NoClassPaddingWalker(sourceFile, this.getOptions()));
  }
}

class NoClassPaddingWalker extends Lint.RuleWalker {
  static ignoreStringsPattern = /([`"'])[\S\s]*?(\1)/g;
  static matchPaddingStartPattern = /class[^{]+{(\s*\n){2,}/;
  static matchPaddingEndPattern = /\n\s*\n\s*}/;

  public visitClassDeclaration(node: ts.ClassDeclaration): void {
    const text = node.getText().replace(NoClassPaddingWalker.ignoreStringsPattern, '');

    if (NoClassPaddingWalker.matchPaddingStartPattern.test(text)) {
      const children = node.getChildren();

      const startNode = children.find((child) => child.getText() === 'class');
      const endNode = children.find((child) => child.getText() === '{');

      this.addFailureAt(startNode.getStart(), endNode.getStart() - startNode.getStart() + 1, Rule.failureMessage);
    }

    if (NoClassPaddingWalker.matchPaddingEndPattern.test(text)) {
      const lastToken = node.getLastToken();
      this.addFailureAt(lastToken.getStart() - 1, lastToken.getEnd(), Rule.failureMessage);
    }

    super.visitClassDeclaration(node);
  }
}
```

Something like this. I don't know if that's the best implementation or if there's a better way.
